test/libsolidity/semanticTests/abiencodedecode/contract_array_v2.sol

```diff
@@ -0,0 +1,12 @@
+pragma experimental ABIEncoderV2;
+contract C {
+  function f(bytes calldata x) public returns (C[] memory) {
+    return abi.decode(x, (C[]));
```

Comment:
Maybe add another similar test with a nested array?

Reply:
Also a test with `abi.encode`?

Reply:
Really, why? I actually didn't even see the point of this test :-). Contracts just have ``address`` as ABI type, there's nothing special about it... and we did have them as arguments to functions already anyways, which uses the same abi decoding functions...
(also I'm lazy and writing tests for nested arrays is really exhausting - and we do have them for other ABI arrays already, which is not really different...)

Reply:
> Also a test with `abi.encode`?

Not really affected by the problem - ``C[] memory x = ...; bytes memory y = abi.encode(x);`` should even work on develop without this and there's no reason to believe anything's wrong with it... and I wouldn't be surprised if we had exactly such tests already... that being said it I'll just add one ;-).

Reply:
Ok, just wanted to be symmetrical with `decode` :p